sdk/src/server-api/sc/protocol/requests/PrepareGameRequest.kt

```diff
@@ -21,17 +21,24 @@ class PrepareGameRequest(
      * The player descriptors default to "Player1" and "Player2".
      *
      * @param gameType type of the game (plugin id)
-     * @param pause whether the game should start paused - default true
+     * @param pause whether the game should start paused
      */
     @JvmOverloads
     constructor(
         gameType: String,
         descriptor1: SlotDescriptor = SlotDescriptor("Player1"),
         descriptor2: SlotDescriptor = SlotDescriptor("Player2"),
-        pause: Boolean = true
+        pause: Boolean = false
     ): this(gameType, arrayOf(descriptor1, descriptor2), pause)
     
-    override fun toString() =
-        "PrepareGameRequest(gameType=$gameType, pause=$pause, slotDescriptors=${slotDescriptors.contentToString()})"
+    override fun equals(other: Any?): Boolean =
+        other is PrepareGameRequest && gameType == other.gameType && pause == other.pause && slotDescriptors.contentDeepEquals(other.slotDescriptors)
+    
+    override fun hashCode(): Int {
+        var result = gameType.hashCode()
+        result = 31 * result + slotDescriptors.contentHashCode()
+        result = 31 * result + pause.hashCode()
+        return result
```

Comment:
Since this is a data class neither `equals` nor `hashCode` should be needed, right? They look generic to me

Reply:
https://github.com/CAU-Kiel-Tech-Inf/backend/blob/10a86759bc28a60e69adcb9278c20029cbc9096e/sdk/src/server-api/sc/protocol/requests/PrepareGameRequest.kt#L34-L42

Reply:
unfortunately it is, IDEA points that out, since Array contents are not considered

Reply:
Because it's a Java Array?
Why not use a different on, like ArrayList?

Reply:
overhead ;)